lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMBase.h

```diff
@@ -339,6 +339,8 @@ class ConvertTritonGPUOpToLLVMPatternBase {
     // Order
     auto inOrder = triton::gpu::getOrder(srcEncoding);
     auto outOrder = triton::gpu::getOrder(resSharedLayout);
+    assert(outVec * (maxPhase - 1) <= srcShape[outOrder[0]] &&
```

Comment:
There's only phases `0, 1, ... maxPhase - 1`, if each one increments by `vec`, then the largest address is `vec * (maxPhase - 1)`

Reply:
The largest address would be `vec * maxPhase - 1` ? I meant the largest starting addresses is `vec * (maxPhase - 1)` , but we actually have accessed addresses until `vec * maxPhase`

Reply:
That's right, I was trying to get away from cases where we have `outVec > 1`, `maxPhase = 1` and `srcShape[outOrder[0]]  = 1`. In that case, we only use address range `[0, 1]`, regardless of what `outVec` is, since we don't really swizzle. Maybe that case should also be illegal, even though in practice the code just works since we don't swizzle.

Reply:
So the condition should be `maxPhase == 1 || vec * maxPhase <= srcShape[outOrder[0]]`, what do you think ?

Reply:
I see. LGTM